src/waveform/widgets/waveformwidgettype.h

```diff
@@ -1,36 +1,41 @@
 #pragma once
 
+#include <QFlags>
+#include <array>
+// required for Qt-Macros
+#include <qobjectdefs.h>
+
 class WaveformWidgetType {
   public:
     enum Type {
         // The order must not be changed because the waveforms are referenced
         // from the sorted preferences by a number.
-        EmptyWaveform = 0,
-        SoftwareSimpleWaveform,      // 1  TODO
-        SoftwareWaveform,            // 2  Filtered
-        QtSimpleWaveform,            // 3  Simple Qt
-        QtWaveform,                  // 4  Filtered Qt
-        GLSimpleWaveform,            // 5  Simple GL
-        GLFilteredWaveform,          // 6  Filtered GL
-        GLSLFilteredWaveform,        // 7  Filtered GLSL
-        HSVWaveform,                 // 8  HSV
-        GLVSyncTest,                 // 9  VSync GL
-        RGBWaveform,                 // 10 RGB
-        GLRGBWaveform,               // 11 RGB GL
-        GLSLRGBWaveform,             // 12 RGB GLSL
-        QtVSyncTest,                 // 13 VSync Qt
-        QtHSVWaveform,               // 14 HSV Qt
-        QtRGBWaveform,               // 15 RGB Qt
-        GLSLRGBStackedWaveform,      // 16 RGB Stacked
-        AllShaderRGBWaveform,        // 17 RGB (all-shaders)
-        AllShaderLRRGBWaveform,      // 18 L/R RGB (all-shaders)
-        AllShaderFilteredWaveform,   // 19 Filtered (all-shaders)
-        AllShaderSimpleWaveform,     // 20 Simple (all-shaders)
-        AllShaderHSVWaveform,        // 21 HSV (all-shaders)
-        AllShaderTexturedFiltered,   // 22 Filtered (textured) (all-shaders)
-        AllShaderTexturedRGB,        // 23 RGB (textured) (all-shaders)
-        AllShaderTexturedStacked,    // 24 Stacked (textured) (all-shaders)
-        AllShaderRGBStackedWaveform, // 26 Stacked (all-shaders)
-        Count_WaveformwidgetType     //    Also used as invalid value
+        Empty = 0,
+        Simple = 5,    // 5  Simple GL
+        Filtered = 7,  // 7  Filtered GLSL
+        HSV = 8,       // 8  HSV
+        VSyncTest = 9, // 9  VSync GL
+        RGB = 12,      // 12 RGB GLSL
+        Stacked = 16,  // 16 RGB Stacked
+        Invalid,       // Don't use! Used to indicate invalid/unknown type, as
+                       // Count_WaveformWidgetType used to.
+    };
+    static constexpr std::array kValues = {
+            WaveformWidgetType::Empty,
+            WaveformWidgetType::Simple,
+            WaveformWidgetType::Filtered,
+            WaveformWidgetType::HSV,
+            WaveformWidgetType::VSyncTest,
+            WaveformWidgetType::RGB,
+            WaveformWidgetType::Stacked,
     };
 };
+
+enum class WaveformWidgetBackend : int {
```

Comment:
I think we have the rule somewhere that you shouldn't specify the underlying type unless you have to.

Reply:
I had to for the static_cast in upgrade, but with your suggested approach there, I will see if I can remove it.

Reply:
yeah. I think you can see how specifying the enums underlying type is a code smell because it allows you to do ugly stuff elsewhere.